src/combinatorics/simple_graph/hasse.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import combinatorics.simple_graph.connectivity
+import data.fin.succ_pred
+import order.succ_pred.relation
+
+/-!
+# The Hasse diagram as a graph
+
+This file defines the Hasse diagram of an order (graph of `covby`, the covering relation) and the
+path graph on `n` vertices.
+
+## Main declarations
+
+* `simple_graph.hasse`: Hasse diagram of an order.
+* `simple_graph.path_graph`: Path graph on `n` vertices.
+-/
+
+open order order_dual relation
+
+namespace simple_graph
+variables (α β : Type*)
+
+section preorder
+variables [preorder α] [preorder β]
+
+/-- The Hasse diagram of an order as a simple graph. The graph of the covering relation. -/
+def hasse : simple_graph α := from_irrefl_rel covby
+
+variables {α β} {a b : α}
+
+@[simp] lemma hasse_adj : (hasse α).adj a b ↔ a ⋖ b ∨ b ⋖ a := iff.rfl
+
+/-- `αᵒᵈ` and `α` have the same Hasse diagram. -/
+def hasse_dual_iso : hasse αᵒᵈ ≃g hasse α :=
+{ map_rel_iff' := λ a b, by simp [or_comm],
+  ..of_dual }
+
+@[simp] lemma hasse_dual_iso_apply (a : αᵒᵈ) : hasse_dual_iso a = of_dual a := rfl
+@[simp] lemma hasse_dual_iso_symm_apply (a : α) : hasse_dual_iso.symm a = to_dual a := rfl
+
+end preorder
+
+section linear_order
+variables [linear_order α]
+
+lemma hasse_preconnected_of_succ [succ_order α] [is_succ_archimedean α] : (hasse α).preconnected :=
+λ a b, begin
+  rw reachable_iff_refl_trans_gen,
+  exact refl_trans_gen_of_succ _ (λ c hc, or.inl $ covby_succ_of_not_is_max hc.2.not_is_max)
+    (λ c hc, or.inr $ covby_succ_of_not_is_max hc.2.not_is_max),
+end
+
+lemma hasse_preconnected_of_pred [pred_order α] [is_pred_archimedean α] : (hasse α).preconnected :=
```

Comment:
Can you prove this from the previous result on `αᵒᵈ`?

Reply:
@Girgias has a proof that connectivity is a graph invariant, but it's not in mathlib yet.

Reply:
I don't have much time until June due to exams, but you can always take inspiration and make me as a co-author on the commit if you use part of https://github.com/Girgias/lean-cw1/blob/master/src/graph-box-product.lean (which was written before mathlib had a def for connectivity tho)

Reply:
ok, this can be changed later until that is added

Reply:
@Girgias By the way, there's a 4-line box product implementation on Zulip due to a question by Yael awhile back: https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Grid.20adjacency.20graph/near/254755265

```lean
def box_product {V W : Type*} (G : simple_graph V) (H : simple_graph W) : simple_graph (V × W) :=
{ adj := λ p q, (G.adj p.1 q.1 ∧ p.2 = q.2) ∨ (p.1 = q.1 ∧ H.adj p.2 q.2),
  symm := λ p q, by simp [and_comm, or_comm, eq_comm, simple_graph.adj_comm],
  loopless := λ p, by simp }
```
If I have time, I might create that co-authored PR (but I don't have very much right now, so don't wait for me).